In [2]:
import pandas as pd
import numpy as np
import statsmodels.api as sm

In [ ]:
# 2

file_path = "gmo_analysis_data.xlsx"
tr = pd.read_excel(file_path, sheet_name="total returns")
rf = pd.read_excel(file_path, sheet_name="risk-free rate")
signal = pd.read_excel(file_path, sheet_name="signals")
tr["date"] = pd.to_datetime(tr["date"])
rf["date"] = pd.to_datetime(rf["date"])
signal["date"] = pd.to_datetime(signal["date"])
tr.set_index("date", inplace=True)
rf.set_index("date", inplace=True)
signal.set_index("date", inplace=True) 

er = tr - rf.values
er.head()

SPY     GMWAX     GMGEX
date                                    
1996-12-31 -0.075002 -0.073804 -0.064710
1997-01-31  0.010316 -0.036735 -0.017022
1997-02-28 -0.042635 -0.029935 -0.039467
1997-03-31 -0.098941 -0.068372 -0.069661
1997-04-30  0.012038 -0.059061 -0.052330

In [6]:
FREQ = 12

def performance_stats(ret):
    """Compute mean, vol, Sharpe (annualized)."""
    mean_ann = ret.mean() * FREQ
    vol_ann = ret.std() * np.sqrt(FREQ)
    sharpe = mean_ann / vol_ann if vol_ann != 0 else np.nan
    return mean_ann, vol_ann, sharpe

def max_drawdown(total_ret):
    wealth = (1 + total_ret).cumprod()
    peak = wealth.cummax()
    dd = (wealth - peak) / peak
    return dd.min()

def tail_risk(ret, total_ret):
    min_ret = ret.min()
    var_5 = ret.quantile(0.05)
    mdd = max_drawdown(total_ret)
    return min_ret, var_5, mdd

def regression_stats(y, x):
    X = sm.add_constant(x)
    model = sm.OLS(y, X).fit()
    alpha_daily = model.params["const"]
    beta = model.params[x.name]
    r2 = model.rsquared
    alpha_ann = alpha_daily * FREQ
    return alpha_ann, beta, r2

In [7]:
# 2.1

samples = {
    "inception_to_2011": er.loc[: "2011-12-31"],
    "2012_to_present":   er.loc["2012-01-01": ],
    "inception_to_present": er
}

print("===== 2.1 PERFORMANCE (GMWAX) =====")
for name, sub in samples.items():
    m, v, s = performance_stats(sub["GMWAX"])
    print(f"\n--- {name} ---")
    print(f"Mean (ann):   {m:.6f}")
    print(f"Vol (ann):    {v:.6f}")
    print(f"Sharpe:       {s:.6f}")

===== 2.1 PERFORMANCE (GMWAX) =====

--- inception_to_2011 ---
Mean (ann):   -0.265291
Vol (ann):    0.131867
Sharpe:       -2.011809

--- 2012_to_present ---
Mean (ann):   -0.123303
Vol (ann):    0.109588
Sharpe:       -1.125154

--- inception_to_present ---
Mean (ann):   -0.197366
Vol (ann):    0.123263
Sharpe:       -1.601179


<span style="color: blue;">

ANS:

The mean became less negative in sub_period 2, while volatility also become less, the Sharpe ratio improve in the sub_period 2, from -2.0 to -1.1.

In [ ]:
#2.2

print("\n===== 2.2 TAIL RISK (GMWAX) =====")
for name, sub in samples.items():
    min_ret, var5, mdd = tail_risk(
        sub["GMWAX"],            # excess return
        tr.loc[sub.index]["GMWAX"]    # total return
    )
    print(f"\n--- {name} ---")
    print(f"Min return:   {min_ret:.6f}")
    print(f"VaR-5%:       {var5:.6f}")
    print(f"Max drawdown: {mdd:.6f}")


===== 2.2 TAIL RISK (GMWAX) =====

--- inception_to_2011 ---
Min return:   -0.193379
VaR-5%:       -0.083767
Max drawdown: -0.293614

--- 2012_to_present ---
Min return:   -0.115577
VaR-5%:       -0.071114
Max drawdown: -0.216795

--- inception_to_present ---
Min return:   -0.193379
VaR-5%:       -0.077006
Max drawdown: -0.293614


<span style="color: blue;">

ANS:

(a) The tail-risk seems high, as Var-5% of monthly data is 7-8% through the sample, and the minimum return is 19% in all the sample, and maximum drawdown is 29.4% in all the sample.

(b) Yes, the second sub-sample seems less ricky in term of tail risk. But it is still quite risky as the Var-5% is 7.1%, maximum drawdown is 22%, and minimum return per month is 11.6%.

In [9]:
# 2.3

print("\n===== 2.3 MARKET EXPOSURE (GMWAX) =====")
for name, sub in samples.items():
    alpha, beta, r2 = regression_stats(
        sub["GMWAX"],     # y
        sub["SPY"]  # x = SPY excess
    )
    print(f"\n--- {name} ---")
    print(f"Alpha (ann):  {alpha:.6f}")
    print(f"Beta:         {beta:.6f}")
    print(f"R²:           {r2:.6f}")



===== 2.3 MARKET EXPOSURE (GMWAX) =====

--- inception_to_2011 ---
Alpha (ann):  -0.094369
Beta:         0.619535
R²:           0.687802

--- 2012_to_present ---
Alpha (ann):  -0.099708
Beta:         0.629431
R²:           0.769300

--- inception_to_present ---
Alpha (ann):  -0.096637
Beta:         0.622398
R²:           0.727059


<span style="color: blue;">

ANS:

GMWAX has around 0.6 Beta, I would not consider it low-beta strategy, and this didn't change across subsamples.

GMWAX does not provide alpha, it has negative alpha across both subsamples.

In [12]:
# 2.4

def analyze_one_fund(fund_name):
    print(f"\n========== RESULTS FOR {fund_name} ==========")

    # 2.1 Performance
    print("\n--- 2.1 PERFORMANCE ---")
    for name, sub in samples.items():
        m, v, s = performance_stats(sub[fund_name])
        print(f"\n{name}")
        print(f"Mean (ann):   {m:.6f}")
        print(f"Vol (ann):    {v:.6f}")
        print(f"Sharpe:       {s:.6f}")

    # 2.2 Tail Risk
    print("\n--- 2.2 TAIL RISK ---")
    for name, sub in samples.items():
        min_ret, var5, mdd = tail_risk(
            sub[fund_name],
            tr.loc[sub.index][fund_name]
        )
        print(f"\n{name}")
        print(f"Min return:   {min_ret:.6f}")
        print(f"VaR-5%:       {var5:.6f}")
        print(f"Max drawdown: {mdd:.6f}")

    # 2.3 Market Exposure Regression
    print("\n--- 2.3 REGRESSION ---")
    for name, sub in samples.items():
        alpha, beta, r2 = regression_stats(
            sub[fund_name],
            er.loc[sub.index]["SPY"]
        )
        print(f"\n{name}")
        print(f"Alpha (ann):  {alpha:.6f}")
        print(f"Beta:         {beta:.6f}")
        print(f"R²:           {r2:.6f}")

# Run comparison
analyze_one_fund("GMWAX")
analyze_one_fund("GMGEX")


========== RESULTS FOR GMWAX ==========

--- 2.1 PERFORMANCE ---

inception_to_2011
Mean (ann):   -0.265291
Vol (ann):    0.131867
Sharpe:       -2.011809

2012_to_present
Mean (ann):   -0.123303
Vol (ann):    0.109588
Sharpe:       -1.125154

inception_to_present
Mean (ann):   -0.197366
Vol (ann):    0.123263
Sharpe:       -1.601179

--- 2.2 TAIL RISK ---

inception_to_2011
Min return:   -0.193379
VaR-5%:       -0.083767
Max drawdown: -0.293614

2012_to_present
Min return:   -0.115577
VaR-5%:       -0.071114
Max drawdown: -0.216795

inception_to_present
Min return:   -0.193379
VaR-5%:       -0.077006
Max drawdown: -0.293614

--- 2.3 REGRESSION ---

inception_to_2011
Alpha (ann):  -0.094369
Beta:         0.619535
R²:           0.687802

2012_to_present
Alpha (ann):  -0.099708
Beta:         0.629431
R²:           0.769300

inception_to_present
Alpha (ann):  -0.096637
Beta:         0.622398
R²:           0.727059

========== RESULTS FOR GMGEX ==========

--- 2.1 PERFORMANCE ---

incepti

<span style="color: blue;">

ANS:

Performance Comparison

GMGEX underperforms GMWAX consistently, showing lower annualized returns and significantly higher volatility.

Both funds have negative Sharpe ratios, but GMGEX’s Sharpe is worse, especially in the post-2012 period.

Tail-Risk Comparison

GMGEX exhibits substantially higher tail risk, including a maximum single-day loss of –66% and maximum drawdowns deeper than –70%.

GMWAX’s worst outcomes are far milder, with drawdowns around –30%.

Market Exposure

GMWAX has a stable low beta (~0.62), consistent with a lower-risk profile.

GMGEX has a higher beta (~0.80) and less stable market exposure (R² drops to ~0.28 after 2012).

Both funds deliver negative alpha, but GMGEX’s alpha deteriorates more severely.

Overall Conclusion

GMWAX is a lower-risk, lower-beta, but low-return strategy.
GMGEX carries much higher risk, larger drawdowns, unstable behavior, and even worse performance.
